extra/mariabackup/xtrabackup.cc

```diff
@@ -3502,15 +3501,55 @@ void backup_wait_for_lsn(lsn_t lsn)
   }
   mysql_mutex_unlock(&recv_sys.mutex);
 }
+#endif
+
+
+/**
+Wait for enough log to be copied.
+@param lsn  log sequence number target
+@return whether log_copying_thread() copied everything until the target lsn
+*/
+static bool backup_wait_for_lsn(lsn_t lsn)
+{
+  if (lsn)
+    msg("Waiting for log copy thread to read lsn " LSN_PF, lsn);
+
+  mysql_mutex_lock(&recv_sys.mutex);
+  ut_ad(!metadata_last_lsn);
+  metadata_last_lsn= lsn ? lsn : 1;
```

Comment:
This could actually be functionally incorrect.

backup_wait_for_lsn() is called after blocking DLL stage_block_ddl() and it might be too early to decide the end_lsn here. It is safer choice to decide the last_lsn at a later stage when we are done with all critical jobs and just need to decide how much more  log to copy to pull the backup forward.